I-am-Erk/CDDA-Tilesets
# [Tools] slicers wrapper

#### Summary
This script allow artists to speedup slicing multitiles process.
It detects tile name, size of resulting tile, and various source files:
  - simple multitile
  - transparent multitile
  - variants for single tile
  - seasonal variants (as well as seasonal transparent variant)

Just drag and drop source folder over script shortcut.


#### Content of the change
1) to slice multitiles python script `slice_multitile.py` requires NumPy module. This module can be used with Python >= 3.7 and < 3.11 So I change `updtset.cmd` so it will install Python 3.10 instead of 3.12 (latest at this moment).
2) `updtset.cmd` will also install `NumPy` module at first run.
3) to keep repository clean from multitiles there is a new line in `.gitignore`
4) added slicemt.py script. 


#### Testing
1) Create a shortcut to the script on your desktop (for example)
![Screenshot 2023-12-19 181936](https://github.com/I-am-Erk/CDDA-Tilesets/assets/6676374/aa7cf8c4-314d-45f5-a936-a7b6c12ea1fc)

2) Put the source files into the folder and drop it in repository
![Screenshot 2023-12-19 182134](https://github.com/I-am-Erk/CDDA-Tilesets/assets/6676374/a868beef-3778-46e8-8377-a04177013b14)

3) Drag and drop 'source' folder over the shortcut
![Screenshot 2023-12-19 182228](https://github.com/I-am-Erk/CDDA-Tilesets/assets/6676374/956acb2c-6282-4195-8397-62447ac29da1)
Script informs you about files taken into account as well as resulting names and sprite numbers. Script will ignore existing json files, so they can be changed without fear.

4) After displaying results script will run 'updtset.cmd' so just press 'Enter' and wait a bit
![Screenshot 2023-12-19 182256](https://github.com/I-am-Erk/CDDA-Tilesets/assets/6676374/fda8f91e-d163-4987-b96a-6c72a873d073)

Now you can update tileset in game and see how it works.



#### Additional information
You may notice shortcut to 'updtset.cmd' on my desktop - it is here just for simple sprites that do not need to slice something. Or if I change json files without touching images.
